framework/src/systems/SystemBase.C

```diff
@@ -107,10 +107,31 @@ SystemBase::SystemBase(SubProblem & subproblem,
     _computing_scaling_jacobian(false),
     _computing_scaling_residual(false),
     _automatic_scaling(false),
-    _verbose(false)
+    _verbose(false),
+    _old_solution_states_needed(0)
 {
 }
 
+void
+SystemBase::setupSolutionStates()
+{
+  // We'll always have 1 state for the current solution
+  _solution_states.resize(_old_solution_states_needed + 1);
+
+  // The first three states (now, old, older) will point to the solutions in the libMesh system,
+  // which is why we are using the "internal" calls to these vectors. _solution_states will then be
+  // the forward facing access to these vectors
+  _solution_states[0] = &solutionInternal();
+  if (_old_solution_states_needed > 0)
+    _solution_states[1] = &solutionOldInternal();
+  if (_old_solution_states_needed > 1)
+    _solution_states[2] = &solutionOlderInternal();
+
+  // Create anything that is past older (state of 3+)
+  for (unsigned int i = 3; i <= _old_solution_states_needed; ++i)
+    _solution_states[i] = &addVector("solution_state_" + std::to_string(i), true, GHOSTED);
```

Comment:
This needs to be modified now that you do lazy setup some object might request 2 older and then another requests 3.  You'll be (re)creating vectors that we already have.